pkg/pod/ttlDestroyHandler.go

```diff
@@ -27,28 +27,39 @@ func (t ttlDestroyHandler) Supports() string {
 
 func (t ttlDestroyHandler) Handle(pod v1.Pod) error {
 	log.Printf("TTL handler invoked: %s", pod.Name)
+
+	creationTimestamp := pod.Labels["im-creationTimestamp"]
 	ttl := pod.Labels["im-ttl"]
-	if ttl != "" && t.ttlBeforeNow(ttl) {
+	if creationTimestamp == "" || ttl == "" {
+		log.Println("No creationTimestamp or TTL found")
+		return nil
+	}
+
+	if t.ttlBeforeNow(creationTimestamp, ttl) {
 		id, err := strconv.ParseUint(pod.Labels["im-id"], 10, 64)
 		if err != nil {
 			return err
 		}
 		payload := struct{ ID uint }{uint(id)}
 		t.producer.Produce(TtlDestroy, payload)
-	} else {
-		log.Println("No TTL found")
 	}
+
 	return nil
 }
 
-func (t ttlDestroyHandler) ttlBeforeNow(ttl string) bool {
+func (t ttlDestroyHandler) ttlBeforeNow(creationTimestamp string, ttl string) bool {
+	creationTimestampInt, err := strconv.ParseInt(creationTimestamp, 10, 64)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+
 	ttlInt, err := strconv.ParseInt(ttl, 10, 64)
 	if err != nil {
 		log.Println(err)
+		return false
 	}
-	ttlTime := time.Unix(ttlInt, 0)
-	loc, _ := time.LoadLocation("UTC")
-	utc := ttlTime.In(loc)
-	now := time.Now()
-	return utc.Before(now)
+
+	ttlTime := time.Unix(creationTimestampInt+ttlInt, 0).UTC()
+	return ttlTime.Before(time.Now())
```

Comment:
Calling `time.Now()` inside functions you want/should test makes testing hard. Time is a side-effect.

If I understand correctly the inspector is not a long-running process rather invoked at certain intervals.

So we could push calling `time.Now()` into main and pass it down. This way we can more easily test that the TTL feature works.

Reply:
Is it really that difficult? Just use Time.Now() in the test when defining the input?

Reply:
You are right when it comes to the before and after cases. You can set `creationTimestamp+ttl` to a value long before or after `time.Now()`. Hitting the exact `time.Now()` might not work or result in a flaky test as the time you get has nanoseconds precision. So achieving `creationTimestamp+ttl` = `time.Now() (called in ttlBeforeNow)` will be hard/impossible.

For me its generally better to treat time as an input like any other. This makes testing easier and more reliable. I'll leave the decision up to you 😋

Reply:
For the sake of complete I'll say the following: I think we should keep it as is for now. It's testable as is and refactoring would only complicate the code unnecessary. But it is a valid point that "now" is hard coded.